taichi/transforms/alg_simp.cpp

```diff
@@ -164,8 +164,14 @@ class AlgSimp : public BasicStmtVisitor {
     if ((fast_math || is_integral(stmt->ret_type)) &&
         irpass::analysis::same_value(stmt->lhs, stmt->rhs)) {
       // fast_math or integral operands: a / a -> 1
-      replace_with_one(stmt);
-      return true;
+      if (stmt->lhs->ret_type->is<PrimitiveType>() &&
```

Comment:
If we just wanna skip TensorType, then better do it this way:
```
if (stmt->ret_type->is<TensorType>() ||
            stmt->rhs->ret_type->is<TensorType>()) {
  return false;
}
```
So we don't overkill. Just in case someone adds another DataType aside from TensorType and PrimitiveType in the future.